Sources/NIOHTTP2/ContentLengthVerifier.swift

```diff
@@ -57,7 +57,15 @@ extension ContentLengthVerifier {
         guard contentLengths.dropFirst().allSatisfy({ $0 == first }) else {
             throw NIOHTTP2Errors.contentLengthHeadersMismatch()
         }
+
         self.expectedContentLength = Int(first, radix: 10)
+
+        guard let expectedLength = self.expectedContentLength else {
+            throw NIOHTTP2Errors.contentLengthHeaderMalformedValue()
+        }
+        if expectedLength < 0 {
+            throw NIOHTTP2Errors.contentLengthHeaderNegative()
+        }
```

Comment:
Note: more of a felling then an actual rule I follow

I usually only assign after the value is fully validated. It doesn't really matter but it signals to me at least that once it is assigned I'm done with it.
```suggestion
        guard let expectedLength = Int(first, radix: 10) else {
            throw NIOHTTP2Errors.contentLengthHeaderMalformedValue()
        }
        if expectedLength < 0 {
            throw NIOHTTP2Errors.contentLengthHeaderNegative()
        }
        self.expectedContentLength = expectedLength
```